arduino/arduino-ide
# #1036 Relaxed the error handling of the core client init

### Motivation


PR To improve the resilience of the IDE2 startup. IDE2 should be able to start although there is a `malformed custom board options` problem with one of the installed cores.

### Change description


This PR relaxes the handling of the received error status messages during the gRPC core client init:
 - the proposed change does not aggressively cancels the gRPC duplex when the first error status is received,
 - improves the logging

### Other information


Closes #1036

How to verify:
 - IDE2 must be able to start and function when the `directories#data` location is empty or does not exists. (The `directories#data` is on `~/Library/Arduino15/`). (Please rename this folder to `/Arduino16` or similar. I deleted it for local checks.)
 - If the IDE2 starts and boards, cores, and libs can be searched, then the PR preserved the async core/lib index update at startup. This is good for performance.
 - Next, follow [the steps](https://github.com/arduino/arduino-ide/issues/1036#issuecomment-1152449530) from the original issue.
 - If the boards, cores, and libs search work but the following error is in the logs, the issues should be fixed:

```
root ERROR Detected an error response during the gRPC core client initialization: code: 9, message: Error loading hardware platform: loading platform release esp8266:esp8266@3.0.0: loading boards: skipping loading of boards esp8266:esp8266:espduino: malformed custom board options
```

### Reviewer checklist
